internal/service/batch/job_queue.go

```diff
@@ -206,14 +244,22 @@ func (r *resourceJobQueue) Update(ctx context.Context, request resource.UpdateRe
 
 	var update bool
 	input := &batch.UpdateJobQueueInput{
-		JobQueue: flex.StringFromFramework(ctx, plan.Name),
+		JobQueue: flex.StringFromFramework(ctx, plan.JobQueueName),
 	}
 
-	if !plan.ComputeEnvironments.Equal(state.ComputeEnvironments) {
-		ceo := flex.ExpandFrameworkStringValueList(ctx, plan.ComputeEnvironments)
-		input.ComputeEnvironmentOrder = expandComputeEnvironmentOrder(ceo)
+	if !plan.ComputeEnvironmentOrder.IsNull() {
+		if !plan.ComputeEnvironmentOrder.Equal(state.ComputeEnvironmentOrder) {
+			flex.Expand(ctx, plan.ComputeEnvironmentOrder, &input.ComputeEnvironmentOrder)
```

Comment:
Can we combine this to make the check a single vs a nested if? No additional checks are done inside the first if.

If you take this suggestion, you'll have a dangling `}`. Suggestions wouldn't let me review it.
```suggestion
	if !plan.ComputeEnvironmentOrder.IsNull() && !plan.ComputeEnvironmentOrder.Equal(state.ComputeEnvironmentOrder) {
			flex.Expand(ctx, plan.ComputeEnvironmentOrder, &input.ComputeEnvironmentOrder)
```

Reply:
good suggestion. retrying tests now

Reply:
cant take the suggestion directly due to the `}`. sorry! tests passed so taking the suggestion